dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/DefaultTrackedEntityAttributeService.java

```diff
@@ -345,6 +345,31 @@ public ProgramTrackedEntityAttribute getProgramTrackedEntityAttribute( Program p
         return programAttributeStore.get( program, trackedEntityAttribute );
     }
 
+    @Override
+    @Transactional( readOnly = true )
+    public Set<TrackedEntityAttribute> getAllTrigramIndexableTrackedEntityAttributes()
+    {
+        return attributeStore.getAllSearchableAndUniqueTrackedEntityAttributes();
+    }
+
+    @Override
+    public void createTrigramIndex( TrackedEntityAttribute trackedEntityAttribute )
```

Comment:
I am wondering if the indexing, analyzing, vacuuming etc deserves its own component/class, and it might not fit well in the service + store structure. This is more of a maintenance/admin concern. Maybe something like `TrackedEntityAttributeTableManager` or similar.

We can then extract out common logic for indexing, vacuuming, analyzing etc for potential other tables/use-cases later. Open for discussion. Makes sense @ameenhere ?